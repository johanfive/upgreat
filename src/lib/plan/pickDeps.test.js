const pickDeps = require('./pickDeps')

const pkgJson = {
  dependencies: {
    p1: '1.1.1',
    p2: '2.2.2',
  },
  devDependencies: {
    p3: '3.3.3',
  },
  peerDependencies: {
    p4: '4.4.4',
  },
}

it('should pick all deps', () => {
  expect(pickDeps(pkgJson)).toMatchSnapshot()
})

it('should respect packageType', () => {
  expect(pickDeps(pkgJson, { packageType: 'dependencies' })).toMatchSnapshot()

  expect(
    pickDeps(pkgJson, { packageType: 'devDependencies' }),
  ).toMatchSnapshot()
})

it('should include by regex', () => {
  expect(pickDeps(pkgJson, { include: 'p3' })).toMatchSnapshot()

  expect(pickDeps(pkgJson, { include: 'p[12]' })).toMatchSnapshot()
})

it('should exclude by regex', () => {
  expect(pickDeps(pkgJson, { exclude: 'p3' })).toMatchSnapshot()

  expect(pickDeps(pkgJson, { exclude: 'p[12]' })).toMatchSnapshot()
})
