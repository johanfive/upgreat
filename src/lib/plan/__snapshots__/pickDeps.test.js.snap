// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should exclude by regex 1`] = `
Array [
  Object {
    "dev": false,
    "name": "p1",
    "version": "1.1.1",
  },
  Object {
    "dev": false,
    "name": "p2",
    "version": "2.2.2",
  },
  Object {
    "dev": false,
    "name": "p4",
    "version": "4.4.4",
  },
]
`;

exports[`should exclude by regex 2`] = `
Array [
  Object {
    "dev": true,
    "name": "p3",
    "version": "3.3.3",
  },
  Object {
    "dev": false,
    "name": "p4",
    "version": "4.4.4",
  },
]
`;

exports[`should include by regex 1`] = `
Array [
  Object {
    "dev": true,
    "name": "p3",
    "version": "3.3.3",
  },
]
`;

exports[`should include by regex 2`] = `
Array [
  Object {
    "dev": false,
    "name": "p1",
    "version": "1.1.1",
  },
  Object {
    "dev": false,
    "name": "p2",
    "version": "2.2.2",
  },
]
`;

exports[`should pick all deps 1`] = `
Array [
  Object {
    "dev": false,
    "name": "p1",
    "version": "1.1.1",
  },
  Object {
    "dev": false,
    "name": "p2",
    "version": "2.2.2",
  },
  Object {
    "dev": true,
    "name": "p3",
    "version": "3.3.3",
  },
  Object {
    "dev": false,
    "name": "p4",
    "version": "4.4.4",
  },
]
`;

exports[`should respect packageType 1`] = `
Array [
  Object {
    "dev": false,
    "name": "p1",
    "version": "1.1.1",
  },
  Object {
    "dev": false,
    "name": "p2",
    "version": "2.2.2",
  },
  Object {
    "dev": false,
    "name": "p4",
    "version": "4.4.4",
  },
]
`;

exports[`should respect packageType 2`] = `
Array [
  Object {
    "dev": true,
    "name": "p3",
    "version": "3.3.3",
  },
  Object {
    "dev": false,
    "name": "p4",
    "version": "4.4.4",
  },
]
`;
